### PR TITLE
[DO NOT MERGE] pinentry: enable libsecret support.

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

**Comment by Cogitri** [comment]
See #4886. This removes the need for libgnome-keyring for pinentry-gnome, but does introduce a dependency on libsecret for all pinentry programs.

**Comment by CameronNemo** [comment]
The libsecret / glib dependency is certainly a downside. But many workstations already use glib somewhere, so it is not the worst dependency to have. Is pinentry commonly used on remote systems?

**Comment by Cogitri** [comment]
> But many workstations already use glib somewhere, so it is not the worst dependency to have.

Yup

> Is pinentry commonly used on remote systems?

Hum, don't think so, but even so I don't think you can really get around the glib dep

**Comment by atweiden** [comment]
> Is pinentry commonly used on remote systems

pinentry is useful for entering gpg private key passwords including on headless systems.

**Comment by atweiden** [comment]
> I don't think you can really get around the glib dep

All a matter of perspective my good man: https://github.com/atweiden/voidpkgs/blob/master/srcpkgs/pinentry/template

**Comment by Cogitri** [comment]
I wasn't talking about pinentry but the entire system

**Comment by atweiden** [comment]
Recently with glib switching to meson [and the problems that caused for static libs](https://gitlab.gnome.org/GNOME/glib/issues/1526), I had reason to look for how to purge glib completely. For context, [atweiden/voidpkgs](https://github.com/atweiden/voidpkgs) aims to provide essential void packages with gui support stripped. Glib is really good at encroaching on that goal, and I just didn't like the whole meson thing enough to look to purge them out. To that end, the only pkgs keeping glib alive in my codebase are avahi, babeltrace, irssi, libimobiledevice, qemu{,-user-static} and rspamd. With the exception of qemu-user-static, most are nonessential and/or have replacements. Remove all those and glib is gone for good :).

A man can dream...

**Comment by Cogitri** [comment]
Hmm...I'm sorry, but I don't really get why you dislike glib upstream so much. They did try to get the meson patches upstream, but they didn't make it into 0.4{8,9}.

Anyway, I don't feel like we should hinder all users from using libsecret (I don't know of any other mainstream-ish distro that doesn't enble this) just because a reaaaaaaaaaaally minor part of Void tries very hard to not have glib on their system

**Comment by atweiden** [comment]
Yes, doubtlessly so. I only seeked to provide an anecdote. Indeed, gpg is useful on headless systems, and there you will use pinentry. Yes, you can escape glib on headless systems :).

**Comment by maxice8** [comment]
Put it behind a default-enabled build_option ? makes it easier for people like @atweiden to avoid it while still providing it for people that want it.

cc @chneukirchen (maintainer)

**Comment by Cogitri** [comment]
Done.

File: srcpkgs/pinentry/template [hunk 1]
```diff
@@ -18,6 +19,9 @@ checksum=68076686fa724a290ea49cdf0d1c0c1500907d1b759a3bcbfbec0293e8f56570
 
 alternatives="pinentry:pinentry:/usr/bin/pinentry-curses"
 
+build_options="libsecret"
+build_options_default="libsecret"
```
Comment (maxice8) [inline srcpkgs/pinentry/template]: libsecret has no entry on common/shlibs, please add

desc_option_libsecret=""

fill it with a very simple explanation of what it is.

grep for `desc_option_` for examples or look into `common/options.description`

**Comment by maxice8** [comment]
On Tue, Dec 25, 2018 at 07:26:12AM -0800, Rasmus Thomsen wrote:
> Cogitri commented on this pull request.
> 
> 
> 
> > @@ -18,6 +19,9 @@ checksum=68076686fa724a290ea49cdf0d1c0c1500907d1b759a3bcbfbec0293e8f56570
>  
>  alternatives="pinentry:pinentry:/usr/bin/pinentry-curses"
>  
> +build_options="libsecret"
> +build_options_default="libsecret"
> 
> How about I add a description to options.description?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/5639#discussion_r243915356

Not enough users

**Comment by Cogitri** [comment]
Alright, done.

**Pull request merged**